internal/controller/controllers/bmh_agent_controller_test.go

```diff
@@ -116,6 +116,105 @@ var _ = Describe("bmac reconcile", func() {
 		})
 	})
 
+	Describe("queue bmh request for cluster deployment", func() {
+		var host *bmh_v1alpha1.BareMetalHost
+		var agent *v1beta1.Agent
+		var agentWithNoMatchingBMH *v1beta1.Agent
+		var clusterDeployment *hivev1.ClusterDeployment
+
+		BeforeEach(func() {
```

Comment:
Try adding a test where there are 2 agents matching the same BMH for a single clusterDeployment. It may be that my logic above is flawed.

Reply:
Your logic above wasn't flawed. Added the test.